### PR TITLE
Added a remains option for digesty vore

### LINKED ISSUE #3802: Suggestion: optional post-digestion remains (skull).
#### Description of the suggestion
A belly mode that allows for some of more longetive parts of prey's body remain (particularly, vore-famous skulls), to allow things like burping it out or depending on vore type, something else. The object itself wouldn't be any dirtier than your average item-that-was-in-belly though, and can be easily gotten rid of. And being optional, you would only have to if you specifically had one made out of your prey's body.

#### Why it should be added?
As long as its completely mutually-agreed and player-controlled, I don't see much more problem with it than any other so far mechanically possible thing. It doesn't allow for any sort of forcing fetish and if people attempt, well, it will be as easy as any other sort of prefbreak and just as obvious.

#### Possibilities of mechanical implementation
If appropriate bellymode is on, then after the body itself is churned away a new indigestable item is added to belly, a skull with description to tell the species of victim and hint at how they met their fate. What is done with it is entirely up to pred, assuming prey is not OOCly against it. Probably an ability to just immediately destroy it by clicking it with your hand, to allow quick and easy way of getting rid of unwanted/accidental ones or cleaning up on mechanical level after the scene played out.

### PR DESCRIPTION
**Pull request by Heroman3003**
Added new Mode Addon: Leave Remains, and a new vore pref under same name.

When addon is active, any prey that has the pref set to ALLOW will leave some bones and a skull after being fully digested. Species without a skeleton (Synths, Alraune, Prommies) do not leave anything even if prefs align. Neither do monkeys to prevent abuse. The remains ((are not digestible ,but they)) can be easily tracelessly destroyed by clicking them while they are in your hand and your intent is set to Harm. Remains also do not carry any DNA, so not usable in investigation stuff (too incomprehensible due to digestion or something).

This is not a feature to allow wearing a belt of skulls or to flood the station with bones. Do not abuse it, as normal preference rules regarding public areas still apply. This is a feature to give a bit more immersion, not a license to prefbreak.

Resolves #3802

EDIT1: also, while existing sprites are ambigously generic enough to fit it all, if somebody makes a sprite set with randomized bone sprites and species-based skull sprites, it'd be greatly appreciated.

EDIT2: taken feedback into account and changed the bones to be digestible and also swallowable with trash eater.

### CODE REVIEW

**Comment by SilverTalismen** [comment]
Honestly I'd suggest making them digestible, since the normal bones you can find on some expediton are digestible, it'd allow people to eat the bones again after or just digest them when they are done with the bones.

**Comment by Heroman3003** [comment]
Main reason it was made indigestible is to allow full item digestion without losing those. In addition to that, the only situation you would even have that option turned on is if you WANT to create the bones. And after you are done, crushing is meant to be equally IC and OOC option of just getting rid of it to avoid prefbreaks/clean up. But I am open to changing that if more people see that as preferrable alternative, just was following logic of "If you want to digest bones, you wouldn't even make them in the first place."

**Comment by SilverTalismen** [comment]
Could always set them up to be protected by the Itemweak digestion setting, it protects limbs and organs in HV aswell as clothing and items and would allow them to be soiled/stained or whatever in the guts of the predator

**Comment by Screemonster** [comment]
I'm not personally bothered by this, but I don't tend to play the janitor and I'm slightly wary of it leading to "oh hey janitor there's a mess in this room and it's your job to clean it up hee hee".

**Comment by Sebbe9123** [comment]
> I'm not personally bothered by this, but I don't tend to play the janitor and I'm slightly wary of it leading to "oh hey janitor there's a mess in this room and it's your job to clean it up hee hee".

9999999% sure doing something like that can get person sent 15 ways to hell if the janitor decides the person who said that just caused a possible Bio-something danger to the station and crew. Lesson to be learned: Don't be a major biggus asshole. Don't make the janitor clean your shit up when that actually isn't the Janitor's job.

**Comment by SpadesNeil** [comment]
This was something that was originally on my hard list of "No" considering it's remains disposal. Personally, not something that bothers me, but actually something I've always wanted to see, and just not been added because I worry about the precedent it might set.

Would you know how to code it so that it prevents you from just dumping the remains out in the open? Basically check by area. Make it so that you can't do that in certain areas.

**Comment by Heroman3003** [comment]
While, in theory, possible, that code will be extreme amount of snowflake and alawys end up just too inflexible. In addition to that, dumping remains in public is not much better than dumping a pile of digested/contaminated clothing, which... Only ever happens in remote areas in maint, medical for handing over or toilets for cleaning. In addition to that result would be way too indiscriminate. For example, one big aspect (and what originally made me make that issue) was skull belching, which is an idea to belch out a skull of prey. Not to keep it or to leave it around, but for purpose of being a horrible tease to next prey/said prey after they are resleeved. That would be destroyed (either via trash eater or harm-click) immedeately. Also, just because area is technically 'open', does not always mean it really is public. Sometimes places like holodeck, library, remote parts of hallways even can be spent a long dragged out scene in with barely any or none interruptions, and no fear of prefbreaking someone.

**Comment by Wickedtemp** [comment]
As long as its treated as a prefs issue if someone leaves remains around in non-secluded areas, I'm fine with it.

**Comment by Screemonster** [comment]
That's a good point. I guess we could use fingerprints, but leave something on the remains so admins can tell who left them. Like, store the pred's name somewhere.

File: code/modules/vore/eating/leave_remains_vr.dm [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+/obj/item/weapon/digestion_remains/skull/unknown
+	desc = "A bleached skull. It looks very weakened. You can't quite tell what species it belonged to."
+	icon_state = "skull"
+
+/obj/item/weapon/digestion_remains/skull/unknown/anthro
+	icon_state = "skull_taj"
+
+/obj/item/weapon/digestion_remains/skull/xenohybrid
+	desc = "A bleached skull. It looks very weakened. Seems like it belonged to something with prolonged head."
```
Comment (ArgobargSoup) [inline code/modules/vore/eating/leave_remains_vr.dm]: "Seems like it belonged to something with an elongated head."

Comment (Heroman3003) [reply code/modules/vore/eating/leave_remains_vr.dm]: Thanks. I was struggling with right word there. Changed.

**Comment by Heroman3003** [comment]
Added pred_ckey and pred_name vars for admin checks

**Comment by Vorrarkul** [comment]
> Species without a skeleton (Synths, Alraune, Prommies) do not leave anything even if prefs align. 

_**I OBJECT**_, my promethean has a skeleton!

**Comment by SpadesNeil** [comment]
>I OBJECT, my promethean has a skeleton!

It's a custom species, then.

**Comment by Heroman3003** [comment]
@Vorrarkul Prometheans do not have skeletons. Canonically and mechanically, they lack bones. If your promethean has a skeleton its not promethean, its a non-official species of slime.

**Comment by Vorrarkul** [comment]
> It's a custom species, then.

If custom species can use promethean sprites as a base then I retract my objection. Otherwise, in seriousness, my promethean keeps a skeleton inside of her because she likes being spoopy. It's not even hers. I would just like to voice support for the ability to represent digesting everything except her bones ingame, if we're going to be implementing bones anyways.

**Comment by Screemonster** [comment]
sounds to me like you want a custom jumpsuit with a sprite that's all bones and shows on the mob as translucent boneyspoop.

**Comment by Wickedtemp** [comment]
Make it glow for bonus spook

**Comment by Vorrarkul** [comment]
...You actually have my attention here. I've been using markings to simulate it up to now.

**Comment by Ketrai** [comment]
It'd be nice if you could toggle the visibility of bones tbh.

**Comment by Heroman3003** [comment]
@Ketrai It was already officially decided in suggestion thread that the whole 'hide from some, show to others' thing will never be a thing on vorestation. Either everyone or noone. And this was deemed mild enough to be everyone.

**Pull request merged**